youtube_dl/utils.py

```diff
@@ -2132,6 +2132,11 @@ def escape_rfc3986(s):
         s = s.encode('utf-8')
     return compat_urllib_parse.quote(s, b"%/;:@&=+$,!~*'()?#[]")
 
+def unescape_rfc3986(s):
+    """Unescape non-ASCII characters as suggested by RFC 3986"""
+    if sys.version_info < (3, 0) and isinstance(s, compat_str):
+        s = s.encode('utf-8')
+    return compat_urllib_parse.unquote(s)
```

Comment:
Nothing to do with RFC 3986.

Reply:
The Percent-Encoding is defined in RFC 3986.

The corresponding method is `escape_rfc3986` so I write a similar method name.

I could improve the docstring citing the Percent-Encoding, if you think it's clearer.

Reply:
Have you even read it?
Percent encoding is a plain mapping of characters to `%XX` representation per se. RFC 3989 determines the set of rules for applying this mapping to URIs, roughly speaking it determines the set of characters that should not be percent encoded so that this set is used in `escape_rfc3986` to fix some invalid URIs to meet the requirements.
What `compat_urllib_parse.unquote` does is simply mapping `%XX` back to character representation for a plain string. This have nothing to do with URIs and with rules determined in RFC 3989.

Reply:
You are right, I've renamed the method an changed the docstring.

Reply:
You don't need dedicated function for that - you already have it `compat_urllib_parse_unquote`.